phidgets_steppers/src/steppers_ros_i.cpp

```diff
@@ -0,0 +1,381 @@
+            position_offset_topicname, rclcpp::QoS(1),
+            std::bind(&StepperSetter::addPositionOffsetCallback, this,
+                      std::placeholders::_1));
+
+    char zero_position_topicname[] = "zero_position00";
+    snprintf(zero_position_topicname, sizeof(zero_position_topicname),
+             "zero_position%02d", (int)channel);
+    zero_position_subscription_ =
+        node->create_subscription<std_msgs::msg::Empty>(
+            zero_position_topicname, rclcpp::QoS(1),
+            std::bind(&StepperSetter::zeroPositionCallback, this,
+                      std::placeholders::_1));
+}
+
+void StepperSetter::setEnabledCallback(const std_msgs::msg::Bool::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setEngaged(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::setTargetPositionCallback(
+    const std_msgs::msg::Float64::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setTargetPosition(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::setVelocityLimitCallback(
+    const std_msgs::msg::Float64::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setVelocityLimit(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::setAccelerationCallback(
+    const std_msgs::msg::Float64::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setAcceleration(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::setCurrentLimitCallback(
+    const std_msgs::msg::Float64::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setCurrentLimit(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::setHoldingCurrentLimitCallback(
+    const std_msgs::msg::Float64::SharedPtr msg)
+{
+    try
+    {
+        steppers_->setHoldingCurrentLimit(channel_, msg->data);
+    } catch (const phidgets::Phidget22Error& err)
+    {
+        // If the data was wrong, the lower layers will throw an exception; just
+        // catch and ignore here so we don't crash the node.
+    }
+}
+
+void StepperSetter::addPositionOffsetCallback(
```

Comment:
I can imagine situations where someone might want to add a position offset, but yes it might not be that useful and it might not hurt to remove it. The problem with making it a parameter is that the zeroPosition service will overwrite any offset that is set at the beginning. I will change it to a service.